src/containers/RecordingView.tsx

```diff
@@ -1,67 +1,44 @@
+import React, { Dispatch, FunctionComponent } from 'react';
 import { connect, MapDispatchToProps, MapStateToProps } from 'react-redux';
 import { RouteComponentProps, withRouter } from 'react-router';
-import { setRecordingInfo } from '../actions';
-import { getRecordingInfo } from '../actions/getRecordingInfo';
+import { WorkspaceInfo } from '../AppContainer';
 import RecordingInfoView from '../components/RecordingInfoView';
-import SortingsView from '../components/SortingsView';
-import { createCalculationPool, HitherContext } from '../extensions/common/hither';
+import { useRecordingInfo } from '../extensions/common/getRecordingInfo';
+import { createCalculationPool } from '../extensions/common/hither';
 import { filterPlugins, Plugins, RecordingSelectionAction } from '../extensions/extensionInterface';
 import sortByPriority from '../extensions/sortByPriority';
-import { getPathQuery } from '../kachery';
 import { RootAction, RootState } from '../reducers';
-import { Recording, RecordingInfo } from '../reducers/recordings';
+import { Recording } from '../reducers/recordings';
 import { Sorting } from '../reducers/sortings';
-import { WorkspaceInfo } from '../reducers/workspaceInfo';
 import { Expandable } from './SortingView';
 
 interface StateProps {
   recording: Recording,
   sortings: Sorting[],
-  workspaceInfo: WorkspaceInfo,
   plugins: Plugins
 }
 
 interface DispatchProps {
-  onSetRecordingInfo: (a: { recordingId: string, recordingInfo: RecordingInfo }) => void
 }
 
 interface OwnProps {
   recordingId: string
+  workspaceInfo: WorkspaceInfo
 }
 
 const calculationPool = createCalculationPool({maxSimultaneous: 6})
 
 type Props = StateProps & DispatchProps & OwnProps & RouteComponentProps
 
-const RecordingView: FunctionComponent<Props> = ({ recordingId, recording, sortings, history, workspaceInfo, onSetRecordingInfo, plugins }) => {
-  const hither = useContext(HitherContext)
-  const { workspaceName, feedUri, readOnly } = workspaceInfo;
-  const [recordingInfo, setRecordingInfo] = useState<RecordingInfo | null>(null)
-  const [errorMessage, setErrorMessage] = useState<string | null>(null)
-
-  useEffect(() => {
-    if ((recording) && (!recordingInfo)) {
-      if (recording.recordingInfo) {
-        setRecordingInfo(recording.recordingInfo)
-        return
-      }
-      getRecordingInfo({ recordingObject: recording.recordingObject, hither }).then((info: RecordingInfo) => {
-        onSetRecordingInfo({recordingId: recording.recordingId, recordingInfo: info})
-      }).catch((err: Error) => {
-        console.error(err)
-        setErrorMessage('Error getting recording info: ' + err.message)
-      })
-    }
-  }, [recording, recording.recordingInfo, recordingInfo, hither, setErrorMessage, onSetRecordingInfo])
+const RecordingView: FunctionComponent<Props> = ({ recordingId, recording, sortings, history, workspaceInfo, plugins }) => {
+  const recordingInfo = useRecordingInfo(recording.recordingObject)
```

Comment:
This is the new mechanism for getting the recording info. Will describe the `useRecordingInfo` and `useSortingInfo` hooks further down.